React/Modules/RCTUIManager.m

```diff
@@ -723,6 +727,20 @@ - (void)_amendPendingUIBlocksWithStylePropagationUpdateForShadowView:(RCTShadowV
   }
 }
 
+- (void)addUIManagerObserver:(id<RCTUIManagerObserver>)observer
+{
+  [_uiManagerObserversLock lock];
```

Comment:
We still have a similar bug with the locks:

1. [UIManager batchDidComplete] -> dispatch_async to the main queue to run the UI blocks
2. Main thread starts running and acquires the observers lock
3. [NativeAnimatedModule batchDidComplete] -> acquires the observers lock